### PR TITLE
feat: Search crud APIs

### PR DESCRIPTION
**Pull request by himank**
## Describe your changes
Adding the following APIs for search service:

- Create Document By Id
- Create Documents
- Create Or Replace Documents
- Update Documents
- Delete Documents
- Delete Documents By Query
- Get Documents

### CODE REVIEW

File: server/v1/openapi.yaml [hunk 1]
```diff
@@ -1897,6 +1897,269 @@ paths:
...
+            description: |-
+                CreateById is used for indexing a single document. The API expects a single document. An "id" is optional
+                 and the server can automatically generate it for you in case it is missing. In cases an id is provided in
+                 the document and the document already exists then that document will not be indexed and an error is returned
+                 with HTTP status code 409.
+            operationId: Search_CreateById
+            parameters:
+                - name: project
+                  in: path
+                  description: Project name whose db is under target to insert documents.
```
Comment (ovaistariq) [inline server/v1/openapi.yaml:2128]: Project name whose search index is under target

File: server/v1/openapi.yaml [hunk 2]
```diff
@@ -1897,6 +1897,269 @@ paths:
...
+                - name: index
+                  in: path
+                  description: index name where to create documents.
+                  required: true
+                  schema:
+                    type: string
+                - name: id
+                  in: path
+                  description: document id.
+                  required: true
```
Comment (ovaistariq) [inline server/v1/openapi.yaml:2141]: `id` is optional?

Comment (himank) [reply server/v1/openapi.yaml:2141]: for this API it is required.

File: server/v1/openapi.yaml (hunk 1 above)
Comment (himank) [reply server/v1/openapi.yaml:2128]: Yes, need to work on updating comments, let me update in my final diff.

File: server/v1/search.proto [hunk 3]
```diff
@@ -56,6 +57,16 @@ message CreateOrUpdateIndexRequest {
   bool only_create = 4;
 }
 
+message IndexSource {
+  // An index can be either managed by user explicitly then the type is set as "user" or the index is backed by Tigris
+  // collection. In case it is backed by Tigris collection the type is "tigris".
+  string Type = 1;
```
Comment (ovaistariq) [inline server/v1/search.proto:63]: Lowercase?

**Review by ovaistariq — APPROVED** [review]

**Review by ovaistariq — APPROVED** [review]

**Pull request merged**